app/helpers/Intersect.js

```diff
@@ -571,3 +572,41 @@ export function disableDebugMode() {
   // Kick off intersection calculations to restore data
   checkIntersect();
 }
+
+async function getPageData(authority, page) {
+  // this function should always return an object.
+  // if there's no data, return an empty object.
+  const cacheKey = authority.internal_id + '|page:' + page.id;
+  let pageData = await GetStoreData(cacheKey);
+  if (!pageData) {
+    if (await shouldDownloadPageData()) {
+      try {
+        pageData = await retrieveUrlAsJson(page.filename);
+        SetStoreData(cacheKey, pageData);
+      } catch (e) {
+        // sometimes the url isn't right, the download will fail
+        pageData = {};
+      }
+    } else {
+      // no cache and can't download, skip this page.
+      pageData = {};
+    }
+  }
+  return pageData;
+}
+
+async function shouldDownloadPageData() {
+  const reduxState = store.getState();
```

Comment:
I believe the store won't be available after we moved the import on this file to an accessor, updating it might require a null check but other than that not terribly different.

Reply:
thank you. I wasn't aware of the change for store accessor. Will update.